library/core/src/ptr/non_null.rs

```diff
@@ -1021,6 +1021,12 @@ impl<T: ?Sized> NonNull<T> {
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
     #[stable(feature = "non_null_convenience", since = "1.80.0")]
     #[rustc_const_stable(feature = "const_intrinsic_copy", since = "1.83.0")]
+    #[requires(count.checked_mul(core::mem::size_of::<T>()).map_or(false, |size| size <= isize::MAX as usize)
```

Comment:
```suggestion
    #[requires(count.checked_mul(core::mem::size_of::<T>()).map_or_else(|| false, |size| size <= isize::MAX as usize)
```
`map_or_else` [is recommended](https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or) when passing the result of a function call.